sdk/openai/azure-openai/samples/chat_completions_oyd.py

```diff
@@ -0,0 +1,73 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: chat_completions_oyd.py
+
+DESCRIPTION:
+    This sample demonstrates how to use your own data with Azure OpenAI Chat Completions.
+
+USAGE:
+    python chat_completions_oyd.py
+
+    Before running the sample:
+
+    pip install "openai" and "azure-identity"
+
+    Set the environment variables with your own values:
+    1) AZURE_OPENAI_ENDPOINT - the endpoint to your Azure OpenAI resource.
+    2) AZURE_OPENAI_CHAT_DEPLOYMENT - the deployment name you chose when deploying your model.
+    3) AZURE_OPENAI_SEARCH_ENDPOINT - the endpoint to your Azure Search resource.
+    4) AZURE_OPENAI_SEARCH_INDEX - the index name you chose when creating your Azure Search index.
+"""
+AZURE_OPENAI_CHAT_DEPLOYMENT="gpt-35-turbo"
+AZURE_OPENAI_ENDPOINT = "AZ_OPENAI_ENDPOINT"
+
+def chat_completion_oyd_studio_viewcode() -> None:
+    import os
+    from openai import AzureOpenAI
+    from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+
+    endpoint = os.getenv(AZURE_OPENAI_ENDPOINT)
+    deployment = os.getenv("AZURE_OPENAI_CHAT_DEPLOYMENT")
+
+    token_provider = get_bearer_token_provider(
+        DefaultAzureCredential(), "https://cognitiveservices.azure.com/.default"
+    )
+
+    client = AzureOpenAI(
+        azure_endpoint=endpoint,
+        azure_ad_token_provider=token_provider,
+        api_version="2024-02-01",
```

Comment:
Should we update this to the June GA?

Reply:
Makes sense, I'll actually update these to pull from `conftest.py` so they get the latest when that's updated.